components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/db/SitePermissionsDatabase.kt

```diff
@@ -0,0 +1,53 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.sitepermissions.db
+
+import android.arch.persistence.room.Database
+import android.arch.persistence.room.Room
+import android.arch.persistence.room.RoomDatabase
+import android.arch.persistence.room.TypeConverter
+import android.arch.persistence.room.TypeConverters
+import android.content.Context
+import mozilla.components.feature.sitepermissions.SitePermissions
+
+/**
+ * Internal database for saving site permissions.
+ */
+@Database(entities = [SitePermissionsEntity::class], version = 1)
+@TypeConverters(StatusConverter::class)
+internal abstract class SitePermissionsDatabase : RoomDatabase() {
+    abstract fun sitePermissionsDao(): SitePermissionsDao
+
+    companion object {
+        @Volatile
+        private var instance: SitePermissionsDatabase? = null
+
+        @Synchronized
+        fun get(context: Context): SitePermissionsDatabase {
+            instance?.let { return it }
+
+            return Room.databaseBuilder(
+                context,
+                SitePermissionsDatabase::class.java,
+                "site_permissions_database"
+            ).build().also { instance = it }
+        }
+    }
+}
+
+@Suppress("unused")
+internal class StatusConverter {
+    private val statusArray = SitePermissions.Status.values()
+
+    @TypeConverter
+    fun toInt(status: SitePermissions.Status): Int {
+        return statusArray.indexOf(status)
+    }
+
+    @TypeConverter
+    fun toStatus(index: Int): SitePermissions.Status {
+        return statusArray[index]
+    }
```

Comment:
This is dangerous since the enum value order is something that is not static, can change and is pretty much not in your control. Someone changing the enum order wouldn't expect other parts of the app to blow up. :)

Since enums can have values. What do you think about defining the int value inside the enum.

Something like:

```kotlin
    enum class Status(
        internal val value: Int
    ) {
        ALLOWED(1),
        BLOCKED(-1),
       NO_DECISION(0)
    }
```

Reply:
👍